Documents/004-Test_Minos_on_ARM_FVP.md

```diff
@@ -0,0 +1,104 @@
+# Test Minos on ARM FVP
+
+Minos uses ARM FVP (Fixed Virtual Platform) for debugging. It has test on latest version of ARM FVP(installed in DS-5_v5.29.3). This guide is based on the DS-5.
+
+You can follow the [ARM FVP Linux Kernel Debugging Concise Manual](https://www.jianshu.com/p/c0a9a4b9569d) for the first view of DS-5 and ARM FVP. Then following the instructions below to debug minos.
+
+## 1. Create a working directory
+
+```shell script
+mkdir ~/minos-workspace/arm-fvp
+```
+
+## 2. Build Minos
+
+   ```shell script
+   cd ~/minos-workspace
+   git clone https://github.com/minosproject/minos.git
+   cd minos
+   make ARCH=aarch64 CROSS_COMPILE=aarch64-linux-gnu- fvp_defconfig
+   make ARCH=aarch64 CROSS_COMPILE=aarch64-linux-gnu- O=0
+   make dtbs
+   ```
+   
+   `O=0` means close the O2 optimization for debug step in.
+
+## 3. Build FVP Kernel
+
+   ``` shell script
+   cd ~/minos-workspace
+   git clone https://github.com/minosproject/linux-marvell.git
+   cd linux-marvell
+   make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- defconfig
+   make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- -j8 Image
+   ```
+
+## 4. Compile ARM Trusted Firmware
+
+   ```shell script
+   cd ~/minos-workspace
+   git clone https://github.com/ARM-software/arm-trusted-firmware.git
+   cd arm-trusted-firmware
+   make CROSS_COMPILE=aarch64-linux-gnu- PLAT=fvp RESET_TO_BL31=1 ARM_LINUX_KERNEL_AS_BL33=1 PRELOADED_BL33_BASE=0xc0000000 ARM_PRELOADED_DTB_BASE=0x83e00000
```

Comment:
PRELOADED_BL33_BASE=### 0xc0000000 ARM_PRELOADED_DTB_BASE=### 0x83e00000

These two address is for old version of minos. need to update to the latest one 0xc0008000 and 0xc3e0000